### PR TITLE
Allow user configs without custom image

### PR DESCRIPTION
**Pull request by ds17f**
I didn't want to build a new image every time I had a different set of sites that I want to front with nginx. This is especially useful if you have dev/test/stage setups and you just want to roll everything out using a common image.

To do this, I leveraged your startup scripts and just added a script that symlinks all configs in a source dir to /etc/nginx/conf.d. That way the certbot config can still live happily there alongside the rest of the config files.

All a user needs to do is put his/her configs into a directory and then mount it using a volume to /etc/nginx/user.conf.d. All *.conf files in that dir will be symlinked on startup.

> NOTE: this is a "redo" of a closed PR.  I worked the bugs out and cleaned it up to be more readable.

### CODE REVIEW

**Comment by ds17f** [comment]
After testing in prod ;)  I discovered that the startup scripts are run too late.  By that point, nginx has crashed because there are no certs for the site files.

To remedy this I moved the functionality as a shell function into the utils and just call it explicitly at the start of the entrypoint.  If the user has no mounted files then a message is printed stating this and execution continues as if I had never modified the code.

File: README.md [hunk 1]
```diff
@@ -43,6 +42,8 @@ services:
             - 443:443/tcp
         environment:
             - CERTBOT_EMAIL=owner@company.com
+        volumes:
+          - ./conf.d:/etc/nginx/user.conf.d
```
Comment (staticfloat) [inline README.md]: Let's put `:ro` at the end of this, to ensure that user configuration is never clobbered by the scripts in this image.

File: example/docker-compose.yml [hunk 2]
```diff
@@ -2,10 +2,12 @@ version: '3'
...
-    build: .
+    image: staticfloat/nginx-certbot
     restart: always
     environment:
       CERTBOT_EMAIL: "your.email@example.com"
     ports:
       - "80:80"
       - "443:443"
+    volumes:
+      - ./conf.d:/etc/nginx/user.conf.d
```
Comment (staticfloat) [inline example/docker-compose.yml]: `:ro` here too

File: src/scripts/util.sh [hunk 3]
```diff
@@ -91,3 +91,22 @@ is_renewal_required() {
     is_finshed_week_sec=$(( ($one_week_sec - $last_renewal_delta_sec) ))
     [ $is_finshed_week_sec -lt 0 ]
 }
+
+# symlinks any *.conf files in /etc/nginx/user.conf.d
+# to /etc/nginx/conf.d so they are included as configs
+# this allows a user to easily mount their own configs
+link_user_configs() {
+    SOURCE_DIR=${1-/etc/nginx/user.conf.d}
```
Comment (staticfloat) [inline src/scripts/util.sh]: Surround in quotes to allow for directories with spaces in them (yes, people that do that bring pain upon themselves, but let's be good neighbors)

File: src/scripts/util.sh [hunk 4]
```diff
@@ -91,3 +91,22 @@ is_renewal_required() {
...
+    TARGET_DIR=${2-/etc/nginx/conf.d}
+
+    echo "symlinking scripts from ${SOURCE_DIR} to ${TARGET_DIR}"
+
+    if [ ! -d "$SOURCE_DIR" ]; then
+        echo "no ${SOURCE_DIR}, nothing to do."
+    else
+        for conf in ${SOURCE_DIR}/*.conf; do
+            echo "symlinking: ${conf}" "${TARGET_DIR}/$(basename ${conf})"
+            ln -s "${conf}" "${TARGET_DIR}/$(basename ${conf})"
```
Comment (staticfloat) [inline src/scripts/util.sh]: I think it would be clearer to just pass `-v` to `ln`, then you can eliminate the `echo` above.

**Review by staticfloat — CHANGES_REQUESTED** [review]

**Comment by staticfloat** [comment]
All minor requests, this is a clear improvement over the previous setup, and best of all, it's backwards-compatible!

File: README.md [hunk 5]
```diff
@@ -11,13 +11,12 @@ This repository was originally forked from `@henridwyer`, many thanks to him for
...
-```Dockerfile
-FROM staticfloat/nginx-certbot
-COPY *.conf /etc/nginx/conf.d/
+Create a config directory for your custom configs:
+```
+mkdir conf.d
 ```
 
-And a `.conf` file such as:
+And a `.conf` file such as in that directory:
```
Comment (karlkeefer) [inline README.md]: rm `such as`

**Comment by karlkeefer** [comment]
Excited about this repo & PR, and agree with the repo owner's review!
Looking forward to using this

**Comment by DDevine** [comment]
I tested this pull request and it worked nicely for me. I look forward to just using this as an image in my docker-compose.yml
I tested it by specifying the git URI in `build:` and mounting my nginx conf in /etc/nginx/user.conf.d

**Comment by ds17f** [comment]
Requests have been addressed.  Sorry to take so long to get back to this.  "Life is what happens when you're busy making other plans."

**Comment by staticfloat** [comment]
Awesome!  Thanks

**Pull request merged**